### PR TITLE
Expose client closed flag publicly

### PR DESCRIPTION
**Pull request by serkantkaraca**
Adding bool IsClosed on EventHubClient, PartitionSender, and PartitionReceiver.

### CODE REVIEW

File: sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/MiscTests.cs [hunk 1]
```diff
@@ -125,5 +125,23 @@ await using (var scope = await EventHubScope.CreateAsync(1))
                 }
             }
         }
+        [Fact]
+        [LiveTest]
+        [DisplayTestMethodName]
+        public async Task ClosingEventHubClientClosesChildEntities()
```
Comment (jsquire) [inline sdk/eventhub/Microsoft.Azure.EventHubs/tests/Client/MiscTests.cs]: nit:  This test is a bit chunky, arguably having responsibility for more than one scenario.   Maybe consider breaking it up into distinct cases:

- Event Hub client marks itself when closed
- Event Hub client  marks a child sender
- Event Hub client marks a child receiver

**Review by jsquire — APPROVED** [review]
LGTM

File: sdk/eventhub/Microsoft.Azure.EventHubs/src/Primitives/ClientEntity.cs [hunk 2]
```diff
@@ -20,6 +20,9 @@ public abstract class ClientEntity
         static int nextId;
         RetryPolicy retryPolicy;
 
+        /// <summary />
+        protected bool closeCalled = false;
```
Comment (pakrym) [inline sdk/eventhub/Microsoft.Azure.EventHubs/src/Primitives/ClientEntity.cs]: Why is this field protected and named this way?

Comment (serkantkaraca) [reply sdk/eventhub/Microsoft.Azure.EventHubs/src/Primitives/ClientEntity.cs]: Sorry, missed to notice your comment. I see your point. I will remove it and set the property setter as protected instead in another PR.

**Pull request merged**